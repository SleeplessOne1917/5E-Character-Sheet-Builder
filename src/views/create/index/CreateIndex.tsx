import ArrowLink from '../../../components/ArrowLink/ArrowLink';
import Link from 'next/link';
import MainContent from '../../../components/MainContent/MainContent';
import classes from './CreateIndex.module.css';
import { initialState as editingCharacterInitialState } from '../../../redux/features/editingCharacter';
import { initialState as editingSpellInitialState } from '../../../redux/features/editingSpell';
import { isEqual } from 'lodash';
import { useAppSelector } from '../../../hooks/reduxHooks';
import { useMemo } from 'react';
import { useRouter } from 'next/router';

const CreateIndex = () => {
	const router = useRouter();
	const editingCharacter = useAppSelector(state => state.editingCharacter);
	const editingSpell = useAppSelector(state => state.editingSpell);
	const viewer = useAppSelector(state => state.viewer);

	const editingCharacterChanged = useMemo(
		() => !isEqual(editingCharacterInitialState, editingCharacter),
		[editingCharacter]
	);

	const editingSpellChanged = useMemo(
		() => !isEqual(editingSpellInitialState, editingSpell),
		[editingSpell]
	);

	return (
		<MainContent testId="create-index">
			<h1>Create</h1>
			<div className={classes.links}>
				<div className={classes['create-link-container']}>
					<ArrowLink
						href={`${router.asPath}/character/race`}
						text={
							editingCharacterChanged
								? `Continue Editing ${
										editingCharacter.name.length > 0
											? editingCharacter.name
											: 'Character'
								  }`
								: 'Create Character'
						}
					/>
				</div>
				{viewer && (
					<>
						<div className={classes['create-link-container']}>
							<ArrowLink
								href={`${router.asPath}/spell`}
								text={`${
									editingSpellChanged ? 'Continue Editing' : 'Create'
								} Spell`}
							/>
						</div>
					</>
				)}
			</div>
			{!viewer && (
				<div className={classes['create-account-blurb']}>
					Create classes, spells, and more by{' '}
					<Link href="/sign-up">
						<a className={classes['create-account-link']}>
							creating an account
						</a>
					</Link>
					.
				</div>
			)}
		</MainContent>
	);
};

export default CreateIndex;
