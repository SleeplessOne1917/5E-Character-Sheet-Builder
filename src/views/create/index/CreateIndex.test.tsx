import * as Router from 'next/navigation';
import * as stories from './CreateIndex.stories';

import { render, screen } from '@testing-library/react';

import { composeStories } from '@storybook/testing-react';

const { Default } = composeStories(stories);

const mockRouter = () => {
	const useRouter = jest.spyOn(Router, 'usePathname');
	useRouter.mockReturnValue('/create');
};

beforeEach(() => {
	mockRouter();
});

it('renders correctly', () => {
	render(<Default />);

	expect(screen.getByTestId('create-index')).toMatchSnapshot();
});
