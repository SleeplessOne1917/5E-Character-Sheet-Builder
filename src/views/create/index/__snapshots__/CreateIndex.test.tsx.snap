// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  class="main"
  data-testid="create-index"
>
  <div
    class="content"
  >
    <h1>
      Create
    </h1>
    <div
      class="links"
    >
      <div
        class="create-link-container"
      >
        <a
          class="arrow-link"
          data-testid="arrow-link"
          href="/character/race"
        >
          Create Character
          <svg
            aria-hidden="true"
            class="link-arrow"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.75 12a.75.75 0 01.75-.75h13.19l-5.47-5.47a.75.75 0 011.06-1.06l6.75 6.75a.75.75 0 010 1.06l-6.75 6.75a.75.75 0 11-1.06-1.06l5.47-5.47H4.5a.75.75 0 01-.75-.75z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      class="create-account-blurb"
    >
      Create classes, spells, and more by
       
      <a
        class="create-account-link"
        href="/sign-up"
      >
        creating an account
      </a>
      .
    </div>
  </div>
</main>
`;
